internal/prompt/editor.go

```diff
@@ -1,68 +1,86 @@
 package prompt
 
 import (
+	"bytes"
 	"fmt"
 	"os"
 	"os/exec"
-	"strings"
+	"runtime"
+
+	"github.com/kballard/go-shellquote"
+)
+
+const (
+	defaultEditor = "vim"
 )
 
-const defaultEditor = "vim"
+var (
+	bom        = []byte{0xef, 0xbb, 0xbf}
+	cliEditors = []string{"vi", "vim", "nano","micro"}
+)
 
-var defaultEditorPrompt = &editorPrompt{defaultEditor: defaultEditor}
+var defaultEditorPrompt = &editorPrompt{editorCmd: getDefaultEditor()}
 
 // CaptureInputViaEditor is the high level function to use in this package in
 // order to capture input from an editor.
 //
 // The arguments have been tailored for our use of strings mostly in the rest
 // of the CLI even though internally we're using []byte.
-func CaptureInputViaEditor(contents, pattern string) (result string, err error) {
-	v, err := defaultEditorPrompt.captureInput([]byte(contents), pattern)
+func CaptureInputViaEditor(contents, pattern string, infoFn func()) (result string, err error) {
+	v, err := defaultEditorPrompt.captureInput([]byte(contents), pattern, infoFn)
 	return string(v), err
 }
 
 type editorPrompt struct {
-	defaultEditor string
+	editorCmd string
```

Comment:
nit: move the functions that have `editorPrompt` as the receiver right below the struct